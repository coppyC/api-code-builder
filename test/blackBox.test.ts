import builder from '../src/main'
import docs from './docs.json'

test('black box', () => {
  const code = builder.buildApi({
    paths: docs.paths as any,
    version: 'ts',
    baseURL: docs.basePath,
    customResponse: 'RESPONSE[\'data\']'
  })
  expect(code).toMatchSnapshot()
})
