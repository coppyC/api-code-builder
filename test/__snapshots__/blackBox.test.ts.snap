// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`black box 1`] = `
"// this file may be overwrite by api code builder, don't change it. Go to api.config.json to config output file
// 本文件可能会被 api code builder 重写覆盖，请勿修改它。配置 api.config.json 来修改生成文件
/* eslint-disable */
import axios from 'axios'

import { AxiosRequestConfig } from 'axios'
axios.defaults.baseURL = '/xxx'

if (!axios.defaults.baseURL)
  axios.defaults.baseURL = '/xxx'

export default {
  admin: {
    /**
     * 登录
     * @explain 开发环境初始账号密码：         xxxx         xxxx
     */
    postLogin(params: AdminPostLoginParams, axiosConfig?: AxiosRequestConfig): Promise<any['data']> {
      const method = 'post'
      return axios('/admin/login', { method, params, ...axiosConfig }) as any
    },
    /** 封禁/解封门店 */
    postShopBanShop(data: ShopBanQo, axiosConfig?: AxiosRequestConfig): Promise<any['data']> {
      const method = 'post'
      return axios('/admin/shop/banShop', { method, data, ...axiosConfig }) as any
    },
    /** 审核门店 */
    postShopCheckShop(data: ShopCheckQo, axiosConfig?: AxiosRequestConfig): Promise<any['data']> {
      const method = 'post'
      return axios('/admin/shop/checkShop', { method, data, ...axiosConfig }) as any
    },
    /** 审核门店资料更新 */
    postShopCheckShopUpdate(data: ShopUpdateCheckQo, axiosConfig?: AxiosRequestConfig): Promise<any['data']> {
      const method = 'post'
      return axios('/admin/shop/checkShopUpdate', { method, data, ...axiosConfig }) as any
    },
    /** 删除服务 */
    deleteShopDeleteService(params: AdminDeleteShopDeleteServiceParams, axiosConfig?: AxiosRequestConfig): Promise<any['data']> {
      const method = 'delete'
      return axios('/admin/shop/deleteService', { method, params, ...axiosConfig }) as any
    },
    /** 删除服务分类 */
    deleteShopDeleteServiceCategory(params: AdminDeleteShopDeleteServiceCategoryParams, axiosConfig?: AxiosRequestConfig): Promise<any['data']> {
      const method = 'delete'
      return axios('/admin/shop/deleteServiceCategory', { method, params, ...axiosConfig }) as any
    },
    /** 设置服务 */
    postShopSaveService(data: ServiceQo[], axiosConfig?: AxiosRequestConfig): Promise<any['data']> {
      const method = 'post'
      return axios('/admin/shop/saveService', { method, data, ...axiosConfig }) as any
    },
    /** 设置服务分类 */
    postShopSaveServiceCategory(data: ServiceCategoryQo[], axiosConfig?: AxiosRequestConfig): Promise<any['data']> {
      const method = 'post'
      return axios('/admin/shop/saveServiceCategory', { method, data, ...axiosConfig }) as any
    },
    /** 门店列表(包括审核列表) */
    getShopShopList(params: AdminGetShopShopListParams, axiosConfig?: AxiosRequestConfig): Promise<CommResultPageVoMerchantBossVo['data']> {
      const method = 'get'
      return axios('/admin/shop/shopList', { method, params, ...axiosConfig }) as any
    },
    /** 申请更新资料门店列表 */
    getShopShopUpdateList(params: AdminGetShopShopUpdateListParams, axiosConfig?: AxiosRequestConfig): Promise<CommResultPageVoShopUpdateVo['data']> {
      const method = 'get'
      return axios('/admin/shop/shopUpdateList', { method, params, ...axiosConfig }) as any
    }
  }
}

export interface AdminPostLoginParams {
  account: string
  captcha: string
  credentials: string
}
export interface AdminDeleteShopDeleteServiceParams {
  serviceId: string
}
export interface AdminDeleteShopDeleteServiceCategoryParams {
  serviceCategoryId: string
}
export interface AdminGetShopShopListParams {
  checkStatus?: number
  currPage?: number
  keyword?: string
  pageSize?: number
}
export interface AdminGetShopShopUpdateListParams {
  currPage?: number
  keyword?: string
  pageSize?: number
  shopId?: string
  updateStatus?: ('0'|'1'|'2')
}
"
`;
